src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,776 @@
+
+    FaceManager & faceManager = mesh.getFaceManager();
+    faceManager.registerExtrinsicData< extrinsicMeshData::FreeSurfaceFaceIndicator >( this->getName() );
+
+    ElementRegionManager & elemManager = mesh.getElemManager();
+
+    elemManager.forElementSubRegions< CellElementSubRegion >( [&]( CellElementSubRegion & subRegion )
+    {
+      subRegion.registerExtrinsicData< extrinsicMeshData::MediumVelocityVp >( this->getName() );
+      subRegion.registerExtrinsicData< extrinsicMeshData::MediumVelocityVs >( this->getName() );
+      subRegion.registerExtrinsicData< extrinsicMeshData::MediumDensity >( this->getName() );
+    } );
+
+  } );
+}
+
+
+
+void ElasticWaveEquationSEM::postProcessInput()
+{
+
+  GEOSX_ERROR_IF( m_sourceCoordinates.size( 1 ) != 3,
+                  "Invalid number of physical coordinates for the sources" );
+
+  GEOSX_ERROR_IF( m_receiverCoordinates.size( 1 ) != 3,
+                  "Invalid number of physical coordinates for the receivers" );
+
+  EventManager const & event = this->getGroupByPath< EventManager >( "/Problem/Events" );
+  real64 const & maxTime = event.getReference< real64 >( EventManager::viewKeyStruct::maxTimeString() );
+  real64 dt = 0;
+  for( localIndex numSubEvent = 0; numSubEvent < event.numSubGroups(); ++numSubEvent )
+  {
+    EventBase const * subEvent = static_cast< EventBase const * >( event.getSubGroups()[numSubEvent] );
+    if( subEvent->getEventName() == "/Solvers/" + this->getName() )
+    {
+      dt = subEvent->getReference< real64 >( EventBase::viewKeyStruct::forceDtString() );
+    }
+  }
+
+  GEOSX_THROW_IF( dt < epsilonLoc*maxTime, "Value for dt: " << dt <<" is smaller than local threshold: " << epsilonLoc, std::runtime_error );
+
+  if( m_dtSeismoTrace > 0 )
+  {
+    m_nsamplesSeismoTrace = int( maxTime / m_dtSeismoTrace) + 1;
+  }
+  else
+  {
+    m_nsamplesSeismoTrace = 0;
+  }
+  localIndex const nsamples = int(maxTime/dt) + 1;
+
+  localIndex numNodesPerElem = 8;
+
+  localIndex const numSourcesGlobal = m_sourceCoordinates.size( 0 );
+  m_sourceNodeIds.resize( numSourcesGlobal, numNodesPerElem );
+  m_sourceConstantsx.resize( numSourcesGlobal, numNodesPerElem );
+  m_sourceConstantsy.resize( numSourcesGlobal, numNodesPerElem );
+  m_sourceConstantsz.resize( numSourcesGlobal, numNodesPerElem );
+  m_sourceIsLocal.resize( numSourcesGlobal );
+
+
+  localIndex const numReceiversGlobal = m_receiverCoordinates.size( 0 );
+  m_receiverNodeIds.resize( numReceiversGlobal, numNodesPerElem );
+  m_receiverConstants.resize( numReceiversGlobal, numNodesPerElem );
+  m_receiverIsLocal.resize( numReceiversGlobal );
+
+  m_displacementxNp1AtReceivers.resize( m_nsamplesSeismoTrace, numReceiversGlobal );
+  m_displacementyNp1AtReceivers.resize( m_nsamplesSeismoTrace, numReceiversGlobal );
+  m_displacementzNp1AtReceivers.resize( m_nsamplesSeismoTrace, numReceiversGlobal );
+  m_sourceValue.resize( nsamples, numSourcesGlobal );
+
```

Comment:
For my own understanding, I see that the `resize` is done in `postProcessInput` for `ElasticWaveEquationSEM`, but in `initializePreSubGroups` in `AcousticWaveEquationSEM`. What is the right place to do it in your opinion? Should you consistify it?

Reply:
HUm I checked and in fact it is duplicated in the Acoustic solver... I think the one we should kept is the one in postProcessInput. We will check it and refactor after.